### PR TITLE
Define a property informing whether an icon was set manually

### PR DESCRIPTION
**Pull request by gmarty**

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -44,6 +43,10 @@ window.GaiaAppIcon = (function(exports) {
     this._hasIcon = false;
     this._hasUserSetIcon = false;
     this._hasPredefinedIcon = false;
+    this._iconUrl = null;
+    this._pendingIconUrl = null;
+    this._pendingIconRefresh = false;
+    this._lastState = 'unavailable';
```
Comment (Cwiiis) [inline script.js]: what's _lastState?

Comment (Cwiiis) [inline script.js]: I don't think 'unavailable' is a valid state, I'd rather it be initialised to null.

File: script.js [hunk 2]
```diff
@@ -286,7 +297,8 @@ window.GaiaAppIcon = (function(exports) {
 
         // Set icon URL on dataset, for testing.
         if (this._pendingIconUrl) {
-          this.dataset.testIconUrl = this._pendingIconUrl;
+          // => querying prop
```
Comment (Cwiiis) [inline script.js]: I'm not sure this is a useful comment.

Comment (gmarty) [inline script.js]: Ooops. :sweat_smile:

**Pull request merged**